src/backend/routers/iep.ts

```diff
@@ -135,6 +135,19 @@ export const iep = router({
     .mutation(async (req) => {
       const { subgoal_id, assignee_id, due_date, trial_count } = req.input;
 
+      const existingTask = await req.ctx.db
+        .selectFrom("task")
+        .where("subgoal_id", "=", subgoal_id)
+        .where("assignee_id", "=", assignee_id)
+        .selectAll()
+        .executeTakeFirst();
+
+      if (existingTask) {
+        throw new Error(
+          "Task already exists: This subgoal has already been assigned to the same para"
+        );
```

Comment:
Throw error instead of making duplicate task. On the frontend we want to try to make sure this doesn't happen.